lib/google_drive/rows.rb

```diff
@@ -55,7 +55,10 @@ def upload_insert(row)
     end
 
     def upload_update(row)
-      @session.request(:put, row.edit_url, :data => row.as_update_xml, :header => {"Content-Type" => "application/atom+xml;charset=utf-8", "If-Match" => "*"})
+      url = row.edit_url
+      xml = row.as_update_xml
+      etag = row.etag
+      @session.request(:put, url, :data => xml, :header => {"Content-Type" => "application/atom+xml;charset=utf-8", "If-Match" => etag})
```

Comment:
I presume if the `etag` aka "row version/lock" is different then we just get a nice friendly error raised by the gem :)